apps/nextjs/src/components/board/sections/dynamic/dynamic-edit-modal.tsx

```diff
@@ -0,0 +1,57 @@
+"use client";
+
+import { Button, ColorInput, Group, Stack, useMantineTheme } from "@mantine/core";
+import { z } from "zod";
+
+import { useZodForm } from "@homarr/form";
+import { createModal } from "@homarr/modals";
+import { useI18n } from "@homarr/translation/client";
+import { dynamicSectionOptionsSchema } from "@homarr/validation";
+
+interface ModalProps {
+  value: z.infer<typeof dynamicSectionOptionsSchema>;
+  onSuccessfulEdit: (value: z.infer<typeof dynamicSectionOptionsSchema>) => void;
+}
+
+export const DynamicSectionEditModal = createModal<ModalProps>(({ actions, innerProps }) => {
+  const t = useI18n();
+  const theme = useMantineTheme();
+
+  const form = useZodForm(dynamicSectionOptionsSchema, {
+    mode: "controlled",
+    initialValues: { ...innerProps.value },
+  });
+
+  return (
+    <form
+      onSubmit={form.onSubmit((values) => {
+        innerProps.onSuccessfulEdit(values);
+        actions.closeModal();
+      })}
+    >
+      <Stack>
+        <ColorInput
```

Comment:
Nitpick: The user must mark the content of the text field and remove it to reset to the default color. Perhaps you could add a ``X`` icon on the right to delete the value?